### ISSUE TITLE
Submodule changes view empty if folder name corresponds to .gitattributes entry

### ISSUE THREAD

**Issue by infotexture** [ForkIssues/Tracker#1366]
I have a repo with multiple submodules, and one of those submodules has a folder name that ends in `.pdf` _(the module contains code for rendering PDFs, it is **not** a PDF file)_.

For all other submodules, the **Changes** view works fine, showing _Old_ and _New_ commit info as expected.

But when I update this submodule, the changes view remains empty, although the **Local Changes** view shows the submodule as modified.

Possibly related: the repo contains a `.gitattributes` entry that tells Git to compare the content of PDFs by treating them as plain text:

```gitattributes
*.pdf   diff=astextplain
```

Not sure if this is what prevents Fork from displaying the submodule changes correctly _(or whether it's treating this as a binary file that can't be compared)_.

Whatever the cause, it seems to be a Fork bug, since other clients like Sourcetree and Tower have no trouble rendering the usual diff of the submodule hashes:

```diff
- Subproject commit a11ab3e0f0b893f3c968fb2512ea0a09f62e6e5c
+ Subproject commit 20d09b7c1dfe0e9961ec78e0dc6038ebdff9412f
```

**Comment by DanPristupov** [comment]
Can you run `git diff --full-index -- foldename.pdf > submodule-diff.txt` in CLI and send me the `submodule-diff.txt` file to support@fork.dev, please?

**Comment by infotexture** [comment]
📧 _On the way…_ 📬

**Comment by DanPristupov** [comment]
The issue is caused by the `.pdf` extension. Fork mistakes the submodule for a PDF file :).

I made a few improvements, could you check that build works properly for you?

https://cdn.fork.dev/prerelease/Fork-2.7.2.dmg

**Comment by infotexture** [comment]
@DanPristupov Thanks, that build solves the problem. 🙏

**Comment by infotexture** [comment]
Thanks for shipping the fix in [2.8](https://github.com/fork-dev/Tracker/milestone/45). 🙏 

I'll close this as resolved.
